File: fizzbuzz.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(int i = 1; i <= n; i++)
using namespace std;

string fizzBuzz(int i) {
    string result;
    result.reserve(8);  // 予めメモリ確保
    if (i % 3 == 0) result += "Fizz";
    if (i % 5 == 0) result += "Buzz";
    return result.empty() ? to_string(i) : result;
}

int main(){
  ios_base::sync_with_stdio(false);
  rep(i,100){
      cout << fizzBuzz(i) << endl;
   }
}

File: fizzbuzz.rs
fn fizzbuzz(count: i32) -> String {
    match (count % 3, count % 5) {
        (0, 0) => "FizzBuzz".to_string(),
        (0, _) => "Fizz".to_string(),
        (_, 0) => "Buzz".to_string(),
        _ => count.to_string(),
    }
}

fn main() {
    (1..=100).for_each(|count| println!("{}", fizzbuzz(count)));
}